### ISSUE TITLE
[Bug Report]Menu component has a problem on the return value of the on-open-change event 

### ISSUE THREAD

**Issue by oyv1cent** [iview/iview#3575]
### Environment
windows10/Chrome 66.0.3359.139/vue 2.5.16

### Reproduction link
[https://jsfiddle.net/yyrzhm46/816/](https://jsfiddle.net/yyrzhm46/816/)
> 
### Steps to reproduce
Menu component non accordion mode on-open-change return value error

### What is expected?
return right value

### What is actually happening?
Return the result that is not in line with the expected results

---
in updateOpenKeys function,openedNames's value is not in line with the expected results。

**Comment by klhsb** [comment]
I met the problem, too

**Comment by WOWHans** [comment]
```
<i-Menu ref="sideMenu" accordion="true" :theme="theme2" active-name="1-2" :open-names="['2']" >
                <Submenu name="1" :key="1">
                    <template slot="title">
                        <Icon type="ios-paper"></Icon>
                        内容管理
                    </template>
                    <Menu-Item name="1-1">文章管理</Menu-Item>
                    <Menu-Item name="1-2">评论管理</Menu-Item>
                    <Menu-Item name="1-3">举报管理</Menu-Item>
                </Submenu>
                <Submenu name="2">
                    <template slot="title">
                        <Icon type="ios-people"></Icon>
                        用户管理
                    </template>
                    <MenuItem name="2-1">新增用户</MenuItem>
                    <MenuItem name="2-2">活跃用户</MenuItem>
                </Submenu>
                <Submenu name="3">
                    <template slot="title">
                        <Icon type="stats-bars"></Icon>
                        统计分析
                    </template>
                    <MenuGroup title="使用">
                        <MenuItem name="3-1">新增和启动</MenuItem>
                        <MenuItem name="3-2">活跃分析</MenuItem>
                        <MenuItem name="3-3">时段分析</MenuItem>
                    </MenuGroup>
                    <MenuGroup title="留存">
                        <MenuItem name="3-4">用户留存</MenuItem>
                        <MenuItem name="3-5">流失用户</MenuItem>
                    </MenuGroup>
                </Submenu>
            </i-Menu>
```
this is ok by use cdn @oyv1cent 
is there any doc?

**Comment by oyv1cent** [comment]
@WOWHans Please set accordion for false, and in non accordion mode test

**Comment by icarusion** [comment]
fixed #3560
